prime-router/src/test/kotlin/azure/SettingsFunctionTests.kt

```diff
@@ -0,0 +1,581 @@
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        val response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can view any organization's settings (getOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting()
+        var response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.getOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot update their organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access user cannot update another organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting(oktaClaimsOrganizationName, mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        val response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.UNAUTHORIZED)
+    }
+
+    @Test
+    fun `test access DHPrimeAdmins can update any organization's settings (updateOneOrganization)`() {
+        val settingsFunction = setupSettingsFunctionForTesting("DHPrimeAdmins", mockFacade())
+        val httpRequestMessage = setupHttpRequestMessageForTesting(HttpMethod.PUT)
+        var response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$organizationName.elr-secondary"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+        response = settingsFunction.updateOneOrganization(
+            httpRequestMessage,
+            "$otherOrganizationName.test-lab-2"
+        )
+        assertThat(response.status).isEqualTo(HttpStatus.OK)
+    }
+
+    @Test
+    fun `test access user cannot get their organization's senders (getSenders-getOneSender)`() {
```

Comment:
The assertion here is wrong as well, a user from an organization can get the senders, but because your not passing in the organization in your getting an unauthorized error.

Reply:
fixed